flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/PartitionITCase.java

```diff
@@ -546,6 +549,60 @@ public void testRangePartitionInIteration() throws Exception {
 		result.collect(); // should fail
 	}
 
+	@Test
+	public void testRangePartitionerOnSequenceDataWithOrders() throws Exception {
```

Comment:
we need at least two more tests here:
- test for range partitioning with key selector and order
- test for range partitioning with a nested key